web/source/osk/oskManager.ts

```diff
@@ -42,10 +42,37 @@ namespace com.keyman.osk {
     dfltX: string;
     dfltY: string;
 
-    // Fields used to store target OSK size when no keyboard has been loaded
-    // If a keyboard has been loaded, the VisualKeyboard values override these.
-    _baseWidth: number;
-    _baseHeight: number;
+    /**
+     * The configured width for this OSKManager.  May be `undefined` or `null`
+     * to allow automatic width scaling. 
+     */
+    private _width: ParsedLengthStyle;
+
+    /**
+     * The configured height for this OSKManager.  May be `undefined` or `null`
+     * to allow automatic height scaling. 
+     */
+    private _height: ParsedLengthStyle;
+
+    /**
+     * The computed width for this OSKManager.  May be null if auto sizing
+     * is allowed and the OSKManager is not currently in the DOM hierarchy.
+     */
+    private _computedWidth: number;
+
+    /**
+    * The computed height for this OSKManager.  May be null if auto sizing
+    * is allowed and the OSKManager is not currently in the DOM hierarchy.
+    */
+    private _computedHeight: number;
```

Comment:
The plain variants will support both relative and absolute specification for an OSK's dimensions:
- percent-based (to fill some portion of the containing element / document)
- pixel-based

In essence, referring to the [OSK-Core design doc](https://docs.google.com/document/d/1uV2IgYv7RQUT318nkJ6Cjwgx8lhGfmtWPqPAD7myfIU/edit#heading=h.cez2bo6bjf6a), this facilitates the different `LayoutHandler` types.
- "static" would use fixed dimensions, while...
- "fill" would use %-based dimensions.

The `_computed` variants will then be used internally to determine the dimensions that the `VisualKeyboard` (along with any other contained elements, like the banner or the desktop header & footer) should fill.

That said, no setters yet, and the only way to 'set' them is explicitly pixel-based at the moment, making `_width` and `_computedWidth` equivalent for now.  This will change after upcoming work.